### PR TITLE
Invite site & onboarding

### PR DESCRIPTION
**Pull request by andrewconner**
This is an unfortunately massive commit due to the templates + CSS + JS.

There are some things in progress (see below), but I thought you guys may want to look at everything else.

A lot of the JS is straight from Alexandru and so can be much improved. I've left much of the HTML/CSS the same as well (from Alexandru), and only fixed specific bugs that I saw. Over time, we'll clean it up.

In summary,
- Authenticated actions now can have an unauthenticated counterpart in some actions. This allows us to handle certain pages differently based on if the user is logged in or not.
- I split out main template parts into reusable templates.
- Users initially are created in a `PENDING` state. We will manually switch them to `ACTIVE` for the near future. By default, all API/JSON actions do not work with pending users. `HtmlAction` optionally includes a flag to allow pending users.
- We're phasing out the blocked experiment. It will be a user state.
- All views are in the `views.html.website` and `views.html.website.onboarding` package.
- All controllers are in the `com.keepit.controllers.website` package.
- We use `UserValue` to determine if a user has accepted the TOS. So, all current users will have to as well.

Not included in this pull request (yet):
- Admin area. Right now, we can switch people from pending to active on their user page, however. The new area will just make it easier to go through people more efficiently.
- Working invite JS + model. I'm having to refactor a lot of what was there originally because it would not be dynamic for anything other than 6 invitations.
- We (I) need to write a pleasant onboarding "what is Kifi" few paragraphs. This also includes friendly invite emails.

I will be updating this today. Right now, everything is admin only. When you log in with dev mode, you will initially be in `pending` state. Go to the admin panel and switch yourself to `active` on your User page.

### CODE REVIEW

File: shoebox/app/assets/javascripts/website/onboarding.js [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+  });
+  
+  
+  //autocomplete
+  
+
+  
+  var users = [
+      {
+        value: "danny-blumenfeld",
```
Comment (gmethvin) [inline shoebox/app/assets/javascripts/website/onboarding.js]: something weird is going on with the indentation here

Comment (andrewconner) [inline shoebox/app/assets/javascripts/website/onboarding.js]: Yeah, it was screwed up everywhere. My current fixes this. Thanks

File: shoebox/public/javascripts/main.js [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+	
+	$("#change_email").on("click", function(){
+		var $this = $(this), newText = $this.data("text");
+		$this.data("text", $this.text());
+		$this.text(newText).toggleClass("confirm");
+		
+		if($this.hasClass("confirm")){
+			$("#email").removeAttr("disabled").trigger("focus");
+		} else {
+			$("#email").attr("disabled", "disabled");
```
Comment (gmethvin) [inline shoebox/public/javascripts/main.js]: for `checked` and `disabled` the recommended way is to use `prop`, like `$('#email').prop('disabled', true)`

File: shoebox/app/com/keepit/controllers/admin/AdminUserController.scala [hunk 3]
```diff
@@ -159,6 +159,18 @@ class AdminUserController @Inject() (
     Ok(Json.obj(experiment -> true))
   }
 
+  def changeState(userId: Id[User], state: String) = AdminJsonAction { request =>
+    val userState = state match {
+      case "active" => UserStates.ACTIVE
+      case "inactive" => UserStates.INACTIVE
+      case "blocked" => UserStates.BLOCKED
+      case "pending" => UserStates.PENDING
+    }
```
Comment (gmethvin) [inline shoebox/app/com/keepit/controllers/admin/AdminUserController.scala:168]: should probably have another case here and send a `BadRequest`

Comment (eishay) [inline shoebox/app/com/keepit/controllers/admin/AdminUserController.scala:168]: would it be safer to do something like 

``` scala
import UserStates._
...
case ACTIVE.value => ACTIVE
case  INACTIVE.value => INACTIVE
...
```

So a typo with the string won't mess up the code

Comment (andrewconner) [inline shoebox/app/com/keepit/controllers/admin/AdminUserController.scala:168]: Good call.

File: shoebox/public/javascripts/main.js (hunk 2 above)
Comment (andrewconner) [inline shoebox/public/javascripts/main.js]: Cool. Most of main.js was left untouched and only added to where I needed to (when functionality was missing). We will definitely be able to clean up a lot of this. Will make the swap.

**Pull request merged**